src/utils/clusterUtils.js

```diff
@@ -221,9 +222,7 @@ export const filterLabels = (labels) => {
   const filteredLabels = {};
 
   for (const key of Object.keys(labels)) {
-    if (
-      key.includes(Constants.RESTRICTED_CLUSTER_LABEL_KEY_SUBSTRING) === false
-    ) {
+    if (validateLabelKey(key).isValid) {
```

Comment:
Using the validator function here, as well, instead of duplicating the validation logic